cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -177,6 +190,8 @@ public void initialize(WorkflowContext context) throws Exception {
       postActions.put(actionSpec.getName(), (PostAction) context.newPluginInstance(actionSpec.getName(),
                                                                                    macroEvaluator));
     }
+
+    WRAPPERLOGGER.info("Pipeline '{}' running", context.getApplicationSpecification().getName());
```

Comment:
I don't think there is 1-1 mapping between the states that we have for programs and the logs for the pipeline. For e.g. program is already in the RUNNING state when the initialize method of the workflow is called, however the pipeline (containing user code) is not yet started, which is why we explicitly log `started` at the beginning of the initialize method and `running` at the end. 

For streaming pipeline again this is completely different where we do not expose state to the user code. So these log statements are strategically placed in the code.